provider/akamai/akamai.go

```diff
@@ -79,84 +86,126 @@ type akamaiZone struct {
 	Zone       string `json:"zone"`
 }
 
-type akamaiRecordsets struct {
-	Recordsets []akamaiRecord `json:"recordsets"`
-}
-
-type akamaiRecord struct {
-	Name  string        `json:"name"`
-	Type  string        `json:"type"`
-	TTL   int64         `json:"ttl"`
-	Rdata []interface{} `json:"rdata"`
-}
-
 // NewAkamaiProvider initializes a new Akamai DNS based Provider.
-func NewAkamaiProvider(akamaiConfig AkamaiConfig) *AkamaiProvider {
-	edgeGridConfig := edgegrid.Config{
-		Host:         akamaiConfig.ServiceConsumerDomain,
-		ClientToken:  akamaiConfig.ClientToken,
-		ClientSecret: akamaiConfig.ClientSecret,
-		AccessToken:  akamaiConfig.AccessToken,
-		MaxBody:      1024,
-		HeaderToSign: []string{
-			"X-External-DNS",
-		},
-		Debug: false,
+func NewAkamaiProvider(akamaiConfig AkamaiConfig, akaService AkamaiDNSService) (provider.Provider, error) {
+	var edgeGridConfig edgegrid.Config
+
+	/*
+		log.Debugf("Host: %s", akamaiConfig.ServiceConsumerDomain)
+		log.Debugf("ClientToken: %s", akamaiConfig.ClientToken)
+		log.Debugf("ClientSecret: %s", akamaiConfig.ClientSecret)
+		log.Debugf("AccessToken: %s", akamaiConfig.AccessToken)
+		log.Debugf("EdgePath: %s", akamaiConfig.EdgercPath)
+		log.Debugf("EdgeSection: %s", akamaiConfig.EdgercSection)
+	*/
```

Comment:
I think we can remove, or do you want to keep that?

Reply:
Prefer to keep. Useful for debugging but don't want always 'on' because they are private auth creds.